### PR TITLE
Java: add unsafe-deserialization support for Flexjson

### PR DESCRIPTION
**Pull request by smowton**
Based on https://github.com/github/codeql/pull/6434

### CODE REVIEW

File: java/ql/src/semmle/code/java/frameworks/Flexjson.qll [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  FlexjsonDeserializerUseMethod() {
+    this.getDeclaringType().getSourceDeclaration().getASourceSupertype*() instanceof
+      FlexjsonDeserializer and
+    this.hasName("use")
+  }
+}
+
+private class FluentUseMethodModel extends SummaryModelCsv {
+  override predicate row(string r) {
+    r = "flexjson;JSONDeserializer;false;use;;;Argument[-1];ReturnValue;value"
```
Comment (Marcono1234) [inline java/ql/src/semmle/code/java/frameworks/Flexjson.qll:39]: This does not consider subtypes (`...;false;...`), but the above `FlexjsonDeserializerUseMethod` does (even though flexjson itself does not seem to have any subtypes).
Should this be made consistent?

File: java/change-notes/2021-08-05-jodd-unsafe-deserialization.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+lgtm,codescanning
```
Comment (atorralba) [inline java/change-notes/2021-08-05-jodd-unsafe-deserialization.md:1]: Committed by accident? It's also in [its own PR](https://github.com/github/codeql/pull/6434/files#diff-7a3164e71b621cbdafe775f66e4049581ff08974a4d2df92cfaf3eb9c4088a3b).

Comment (smowton) [reply java/change-notes/2021-08-05-jodd-unsafe-deserialization.md:1]: This PR is based on that one; in this PR please only review the top commits relating to Flexjson

**Review by atorralba — DISMISSED** [review]

**Comment by atorralba** [comment]
The Flexjson part LGTM.

**Review by aschackmull — APPROVED** [review]

**Pull request merged**